libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/RustMatrixClientFactory.kt

```diff
@@ -71,21 +71,18 @@ class RustMatrixClientFactory @Inject constructor(
         )
     }
 
-    internal fun getBaseClientBuilder(sessionPath: String): ClientBuilder {
+    internal fun getBaseClientBuilder(sessionPath: String, passphrase: String?): ClientBuilder {
         return ClientBuilder()
             .sessionPath(sessionPath)
+            .passphrase(passphrase)
+            .slidingSyncProxy(AuthenticationConfig.SLIDING_SYNC_PROXY_URL)
             .userAgent(userAgentProvider.provide())
             .addRootCertificates(userCertificatesProvider.provides())
+            .autoEnableBackups(true)
+            .autoEnableCrossSigning(true)
             .serverVersions(listOf("v1.0", "v1.1", "v1.2", "v1.3", "v1.4", "v1.5"))
-            .let {
-                // Sadly ClientBuilder.proxy() does not accept null :/
-                // Tracked by https://github.com/matrix-org/matrix-rust-sdk/issues/3159
-                val proxy = proxyProvider.provides()
-                if (proxy != null) {
-                    it.proxy(proxy)
-                } else {
-                    it
-                }
+            .apply {
+                proxyProvider.provides()?.let { proxy(it) }
```

Comment:
Are you sure that setting the proxy like this is similar to how it was done before?
According to https://github.com/matrix-org/matrix-rust-sdk/blob/main/bindings/matrix-sdk-ffi/src/client_builder.rs#L382, I am afraid it's not.

Reply:
Sorry, I don't really see the issue so I may have overlooked something. I think the contents of the `let` block and this are equivalent, just a bit more concise?

Reply:
I mean, they were both applied to a `ClientBuilder` and that method in the SDK hasn't changed in the last year or so.

Reply:
apply will not return the result of `proxy(it)` but will return the receiver.
So I am not sure that the receiver and the returned value of `proxy` is the same. Maybe this is demoing what I am thinking, you can see that `c` and `d` are not the same object:

<img width="902" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/eeaff95a-5556-4b0a-b38f-1db86efb69d8">

<img width="914" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/ab964be2-b1a3-49ce-ad81-a3005f3d7325">

You can also test with this:

<img width="835" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/a8405343-e8f7-4d1f-87fc-db4b2338bd0e">

The application will not be able to restore the session.

Reply:
Oh, you're right, thanks. This is the 2nd time this has bit me with Rust Builders, I'll have to make a mental note about it.

Reply:
Should be fixed in https://github.com/element-hq/element-x-android/pull/3068/commits/42ae60b54d6862659bb7d31f2b11b0c28ee1d79b